### ISSUE TITLE
moodle: Add way to add plugins

### ISSUE THREAD

**Issue by fernsehmuell** [NixOS/nixpkgs#112488]
**Describe the bug**
It should be possible to install optional moodle-plugins

The wordpress service has an option called plugins (https://github.com/NixOS/nixpkgs/blob/nixos-20.09/nixos/modules/services/web-apps/wordpress.nix)

It would be nice if we could have something similar for moodle to install for example the Jitsi Plugin.

regards
Udo Sauer

**Comment by Kloenk** [comment]
Moodle has the plugins inputs, where you can give plugins into the derivation.

~~~nix
{ pkgs, ... }:
let
  bbb = pkgs.moodle-utils.buildMoodlePlugin {
    name = "bigbluebuttonbn";
    src = pkgs.fetchzip {
      name = "bbb-moodle-plugin";
      url =
        "https://moodle.org/plugins/download.php/23293/mod_bigbluebuttonbn_moodle310_2019042011.zip";
      sha256 =
        "12aclcjm3wq8lz7ks3nk1pb0is2a2wljy7db6qjc2v4hjnybhrlq";
      extraPostFetch = ''
        echo fix perm: $out
        chmod a-w -R $out
      '';
    };
    pluginType = "mod";
  };
in pkgs.moodle.override { plugins = [ bbb ]; }
~~~

**Comment by fernsehmuell** [comment]
I just realised, that i can only use 6 types of plugins (defined in default.nix: mod, theme, block, question, course and report. But there are a lot more than that: (see https://docs.moodle.org/dev/Plugin_types). I want to install a "local" plugin. Can that be done somehow?

**Comment by Kloenk** [comment]
I think you have the extend the nixpkgs function for that. Did not see the list as I did the pr.

**Comment by Kloenk** [comment]
This should be fixed with #93104, if not please reopen the issue.
